From 1e2f28a5bdf96a4006dfa00487b279caea5e34c8 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 17 Jan 2011 13:51:06 +0800
Subject: [PATCH] ENGR00137991-1 MSL mxc edid: add cea ext block parser

add cea ext block parser.
it provide more video modes and the info of device HDMI compatible.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 arch/arm/plat-mxc/include/mach/mxc_edid.h |   14 +++++++++++---
 1 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/arch/arm/plat-mxc/include/mach/mxc_edid.h b/arch/arm/plat-mxc/include/mach/mxc_edid.h
index 155ec36..f36a1d2 100644
--- a/arch/arm/plat-mxc/include/mach/mxc_edid.h
+++ b/arch/arm/plat-mxc/include/mach/mxc_edid.h
@@ -1,5 +1,5 @@
 /*
- * Copyright 2009-2010 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2009-2011 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -26,7 +26,15 @@
 #ifndef MXC_EDID_H
 #define MXC_EDID_H
 
-int read_edid(struct i2c_adapter *adp,
-	       unsigned char *edid);
+struct mxc_edid_cfg {
+	bool cea_underscan;
+	bool cea_basicaudio;
+	bool cea_ycbcr444;
+	bool cea_ycbcr422;
+	bool hdmi_cap;
+};
+
+int mxc_edid_read(struct i2c_adapter *adp, unsigned char *edid,
+	struct mxc_edid_cfg *cfg, struct fb_info *fbi);
 
 #endif
-- 
1.5.4.4

